plugin.rb

```diff
@@ -104,6 +104,20 @@ class Engine < ::Rails::Engine
 
     reaction_users_counting_as_like.flatten!
 
+    # Used to remove historical like data in the filter below.
+    previous_main_reaction_like_ids =
+      object
+        .post_actions
+        .where(
+          "post_actions.post_action_type_id = :like AND
+            post_actions.post_id IN (
+              #{DiscourseReactions::PostActionExtension.post_action_with_reaction_user_sql}
+            )",
+          like: PostActionType.types[:like],
+          valid_reactions: [DiscourseReactions::Reaction.main_reaction_id],
+        )
+        .pluck(:id)
```

Comment:
Also, if this is being added to the `PostSerializer`, it will still cause an N+1 if we are serializing an array of `Post` records.

Reply:
Ah yeah you're right :/ It's just a "less bad" N1. Will find a way to fix before  merging.